lib/streamlit/user_info.py

```diff
@@ -48,14 +48,193 @@
 
 @gather_metrics("login")
 def login(provider: str | None = None) -> None:
-    """Initiate the login for the given provider.
+    """Initiate the login flow for the given provider.
+
+    This command redirects the user to an OpenID Connect (OIDC) provider. After
+    the user authenticates their identity, they are redirected back to the
+    home page of your app. This creates a new session where the user's identity
+    is available through |st.experimental_user|_. Call ``st.logout()`` to
+    remove the user's information from ``st.experimental_user`` and start a new
+    session.
+
+    You can use any OpenID Connect (OIDC) provider, including GitHub, Google,
+    Microsoft, Okta, and more. You must configure the provider through secrets
+    management. Although OIDC is an extension of OAuth 2.0, you can't use
+    generic OAuth providers. Furthermore, you can only access the user's
+    identity information and not their underlying OAuth token. Therefore, this
+    command will not allow your app to act on behalf of a user in a secure
+    system.
+
+    For all providers, there are two common settings, ``auth.redirect_uri`` and
+    ``auth.cookie_secret``, which you must specify in the ``[auth]`` dictionary
+    in ``secrets.toml``. Other settings must be defined as described in the
+    ``provider`` parameter.
+
+    - ``auth.redirect_uri`` is your app's URL with the pathname
+      ``/oauth2callback``.
+    - ``auth.cookie_secret`` should be a strong, randomly generated secret.
+
+    .. Important::
+        - You must install ``Authlib>=1.3.2`` to use this command.
+        - Your authentication configuration is dependent on your host location.
+          When you deploy your app, remember to update your ``redirect_uri``
+          both within your app and within your provider.
+        - Streamlit will automatically enable CORS and XSRF protection when you
+          use ``st.login()``, overriding configuration options in
```

Comment:
"""... and XSRF protection when you use ``st.login()``""" it would be more correct to say """when Auth configured via ``secrets.toml``""""